plugins/window_size/windows/window_size_plugin.cpp

```diff
@@ -0,0 +1,195 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "window_size_plugin.h"
+
+#include <ShellScalingApi.h>
+#include <windows.h>
+
+#include <VersionHelpers.h>
+#include <flutter/flutter_view.h>
+#include <flutter/method_channel.h>
+#include <flutter/plugin_registrar_windows.h>
+#include <flutter/standard_method_codec.h>
+#include <memory>
+#include <sstream>
+
+namespace {
+
+using flutter::EncodableList;
+using flutter::EncodableMap;
+using flutter::EncodableValue;
+
+// See window_size_channel.dart for documentation.
+const char kChannelName[] = "flutter/windowsize";
+const char kGetScreenListMethod[] = "getScreenList";
+const char kGetWindowInfoMethod[] = "getWindowInfo";
+const char kSetWindowFrameMethod[] = "setWindowFrame";
+const char kFrameKey[] = "frame";
+const char kVisibleFrameKey[] = "visibleFrame";
+const char kScaleFactorKey[] = "scaleFactor";
+const char kScreenKey[] = "screen";
+
+const double kBaseDpi = 96.0;
+
+// Returns the serializable form of |frame| expected by the platform channel.
+EncodableValue GetPlatformChannelRepresentationForRect(const RECT &rect) {
+  return EncodableValue(EncodableList{
+      EncodableValue(static_cast<double>(rect.left)),
+      EncodableValue(static_cast<double>(rect.top)),
+      EncodableValue(static_cast<double>(rect.right - rect.left)),
+      EncodableValue(static_cast<double>(rect.bottom - rect.top)),
+  });
+}
+
+// Extracts information from monitor |monitor| and returns the
+// serializable form expected by the platform channel.
+EncodableValue GetPlatformChannelRepresentationForMonitor(HMONITOR monitor) {
+  if (!monitor) {
+    return EncodableValue();
+  }
+
+  MONITORINFO info;
+  info.cbSize = sizeof(MONITORINFO);
+  GetMonitorInfo(monitor, &info);
+  UINT dpi_x, dpi_y;
+  GetDpiForMonitor(monitor, MDT_EFFECTIVE_DPI, &dpi_x, &dpi_y);
```

Comment:
From https://docs.microsoft.com/en-us/windows/win32/api/shellscalingapi/nf-shellscalingapi-getdpiformonitor#remarks, this api is not DPI aware. We should check for the awareness level and use  GetDpiForWindow instead. We would need a way to get the DPI per awareness setting. I started something like this here https://github.com/flutter/engine/pull/12735/files#diff-61a7e1c427ae9b307bd89eb9b20229a2R30. I can take it over again and land it before this PR.

Reply:
Yes, I saw that but was confused since it's what the Runner code is currently using, and that's clearly reacting to DPI. And this gives me expected values on my laptop. And we can't use GetDpiForWindows because this is monitor-level information—we don't have a window, and there may not even be one (unless there's some kind of meta-window on each monitor automatically?)

I figured being no worse than the Runner currently is seemed like a reasonable starting place, and then we can revisit this when we've cleaned up the Runner and engine to do DPI coherently.

Reply:
Sounds good, we should revisit dpi awareness soon anyway